app/lib/hooks/useFeatureFlags.ts

```diff
@@ -0,0 +1,29 @@
+import { IS_PREVIEW_BUILD, IS_PRODUCTION } from "lib/constants";
+import { useMemo } from "react";
+
+type FeatureFlags = {
+  testFeatureFlag: boolean;
+  redemptions: boolean;
+};
+
+const DEVELOPMENT: FeatureFlags = { testFeatureFlag: true, redemptions: true };
+const PREVIEW: FeatureFlags = { testFeatureFlag: false, redemptions: true };
+// const STAGING: FeatureFlags = { testFeatureFlag: false, redemptions: false };
+const PRODUCTION: FeatureFlags = { testFeatureFlag: false, redemptions: true };
+
+const getFeatures = (): FeatureFlags => {
+  if (IS_PREVIEW_BUILD) return PREVIEW;
+  if (IS_PRODUCTION) return PRODUCTION;
+
+  return DEVELOPMENT;
+};
+
+export const useFeatureFlags = () => {
```

Comment:
A hook would be useful if we wanted to update these on-the-fly, like based on some state that can change at runtime.
But our prod builds only take 3 mins or less, and we have preview links for everything, so static is fine

Also the downside of the hook is we can't use it outside of a react context. We may need to flag things inside of actions, utils and getStaticProps (for redirects)

So I'd just go with a simple config object. The only 3 states we need are
1. Enable everywhere
2. Only enable on preview + local (hide on prod)
3. Only enable on local

```ts
const featureFlags = {
  selectiveRetirement: true, // enable everywhere
  aggregatorV2: !IS_PRODUCTION, // enable on preview and local
  redemptionUI: IS_LOCAL_DEVELOPMENT, // enable on local
}
```

hold up: we have the complexity that env vars don't exist in `lib`. But we don't want two separate configs.

So I think the best way to do this is to reexport at runtime
```ts
// in `lib`
const getFeatureFlags = (params: { isProduction: boolean, isLocal: boolean }) => {
  const featureFlags = {
      selectiveRetirement: true, // enable everywhere
      aggregatorV2: !params.isProduction, // enable on preview and local
      redemptionUI: isLocal, // enable on local (hide on preview and prod)
  };
  return featureFlags
}

// in `site/lib/featureFlags` and `app/lib/featureFlags`
export const featureFlags = getFeatureFlags({ isProduction: IS_PRODUCTION, isLocal: IS_LOCAL_DEVELOPMENT });

// in react component
import { featureFlags } from "lib/featureFlags"
if (featureFlags.redemptionUi) return <RedemptionUI />
```

Reply:
Love this, it's way more simple than the original implementation.

To improve on this suggestion further, we can drop the `isLocal` param altogether.
I reakon its redundant to enable it on local dev only. I consider preview builds a "development" environment too.